cpp/src/parquet/column_reader.cc

```diff
@@ -1138,6 +1146,14 @@ int64_t TypedColumnReaderImpl<DType>::ReadBatchSpaced(
   return total_values;
 }
 
+template <typename DType>
```

Comment:
is this replicating the change from the other PR?

Reply:
If so, should we close the other one?

Reply:
I will try to drop the extra commits here.